app/javascript/mastodon/components/autosuggest_textarea.js

```diff
@@ -79,6 +79,7 @@ class AutosuggestTextarea extends ImmutablePureComponent {
     }
 
     // auto-resize textarea
+    e.target.style.height = 'auto';
```

Comment:
Surprisingly, without this line the height of the text area grows but it will not shrink.